File: Leetcode/1464. Maximum Product of Two Elements in an Array/solution.go
func maxProduct(nums []int) int {
	val_1, val_2 := 0, 0

	for _, num := range nums {
		if num > val_1 {
			val_2 = val_1
			val_1 = num
		} else if num > val_2 {
			val_2 = num
		}
	}

	return (val_1 - 1) * (val_2 - 1)
}

File: Leetcode/1464. Maximum Product of Two Elements in an Array/solution.rs
impl Solution {
    pub fn max_product(nums: Vec<i32>) -> i32 {
        let mut val_1 = 0;
        let mut val_2 = 0;
        
        for num in nums {
            if num > val_1 {
                val_2 = val_1;
                val_1 = num;
            }
            else if num > val_2{
                val_2 = num;
            }
        }
        return (val_1 - 1) * (val_2 - 1)
    }
}